cpp/src/Ice/Network.cpp

```diff
@@ -1519,6 +1521,21 @@ IceInternal::getRecvBufferSize(SOCKET fd)
     return sz;
 }
 
+bool
+IceInternal::hasBytesAvailable(SOCKET fd) noexcept
+{
+#ifdef _WIN32
+    unsigned long bytesAvailable = 0;
+    int rs = ioctlsocket(fd, FIONREAD, &bytesAvailable);
+    assert(rs == 0);
+#else
+    int bytesAvailable = 0;
+    int rs = ioctl(fd, FIONREAD, &bytesAvailable);
+    assert(rs == 0);
+#endif
+    return rs == 0 && bytesAvailable > 0;
```

Comment:
If ioctl fails, it's most likely because of a bug in our code, like we're somehow closing the socket while the idle check is running. And I want to get an assert in debug builds when this happens.

In a release build, if it fails, I'd rather not return a false positive. Failure of this call doesn't mean "bytes are available".

Reply:
For other code, we typically through an exception for unexpected failures.